### DISCUSSION TITLE
MongoDB Filter Boolean not Equal True Not Working Properly

### DISCUSSION THREAD

**Discussion by indicozy** [prisma/prisma#13961] [Q&A] [answered]
Hi everyone!

I'm trying to create a query with boolean, where boolean value can be `null`, `false`, but not equal to `true`.

### Query
```js
  const submission = await prisma.submission.findMany({
    where: {
      reviewSuccess: {
        not: true, // HERE
      },
      blockId,
      block: {
        reviewable: {
          equals: true,
        },
      },
      success: {
        equals: true,
      },
      peerReviews: {
        none: {
          studentId: student.id,
        },
      },
      studentId: {
        not: student.id,
      },
    },
    include: {
    // some stuff
},
    orderBy: [
      {
        updatedAt: 'asc',
      },
    ],
  });
```

Here's query's return with removed `reviewSuccess` filter:

```
[
  {
    "id": "62b2fa8b24278acf04c93878",
    "createdAt": "2022-06-22T11:18:35.601Z",
    "updatedAt": "2022-06-22T11:18:35.601Z",
    "studentId": "62b2fa4c24278acf04c93876",
    "blockId": "62b2edea641f7886c72e55de",
    "data": {
      "code": "x = 20",
      "codeOutput": {
        "stdout": "20\n",
        "stderr": "",
        "status": true
      }
    },
    "success": true,
    "reviewSuccess": false, // HERE
    "student": {
      "User": {
        "image": null,
        "Profile": {
          "username": "feminine-plum-blackcapped-chickadee",
          "image": null
        }
      }
    },
    "_count": {
      "peerReviews": 0
    }
  }
]
```

### Question

But it does not return documents where `reviewSuccess` is `false` or `null`. I'm not sure what to do, do you have any idea?

### Schema
```
model Submission {
  id        String         @id @default(auto()) @map("_id") @db.ObjectId
  createdAt DateTime       @default(now())
  updatedAt DateTime       @updatedAt
  student   Student        @relation(fields: [studentId], references: [id])
  studentId String         @db.ObjectId
  block     Block          @relation(fields: [blockId], references: [id])
  blockId   String         @db.ObjectId
  data      SubmissionData
  success   Boolean

  peerReviews   PeerReview[]
  reviewSuccess Boolean      @default(false)
  @@unique([studentId, blockId])
}
```

**Comment by indicozy** [comment, marked as answer]
Alright, after an hour of brainstorming and reboot everything works as expected. Closing the discussion

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
